### DISCUSSION TITLE
Reduce polling

### DISCUSSION THREAD

**Discussion by ktreptow** [concourse/concourse#6358] [Help & Support] [answered]
Hey everybody, 
excessive resource polling can, depending on the infrastructure, get pretty expensive pretty fast. My goal is to use webhooks as build triggers whenever possible, set `check_every` for specific resources/resource-types and use environmental default values to check everything else, that is not explicitly declared every 10 minutes or so. Unfortunately I'm still not sure which variables to use and how they affect each other. I found out about the following ones so far: 

```
CONCOURSE_RESOURCE_CHECKING_INTERVAL
CONCOURSE_RESOURCE_TYPE_CHECKING_INTERVAL
CONCOURSE_LIDAR_SCANNER_INTERVAL
CONCOURSE_LIDAR_CHECKER_INTERVAL
```
I already set them to the values I'd like but somehow they don't seem to work as i expected. Is there any documentation on all of them, their meaning and how i should set them when using them together? I wasn't able to find anything that answered my questions. 

Also I found [this comment](https://github.com/concourse/concourse/issues/2050#issuecomment-367335072
) but can't really say if that's relevant.

Thank you very much in advance!

**Comment by taylorsilva** [comment, marked as answer]
These are only used if not specified in the resource's config: https://concourse-ci.org/resources.html#schema.resource.check_every
If a `check_every` is specified in the resource's config then that value will override what you've set on the web node.
```
CONCOURSE_RESOURCE_CHECKING_INTERVAL
CONCOURSE_RESOURCE_TYPE_CHECKING_INTERVAL
```

The lidar flags you can leave as-is probably. They tell Concourse how often to check if any checks do need to happen. In general I would not change those.

Hope that helps clear up whatever weird behaviour you're seeing!

**Reply by ktreptow** [reply]
Hi, thanks for your reply! 
Unfortunately it seemingly worker for some time, then stopped polling completely. I used the following configuration: 

```
CONCOURSE_RESOURCE_CHECKING_INTERVAL = "10m"
CONCOURSE_RESOURCE_TYPE_CHECKING_INTERVAL = "10m"
CONCOURSE_LIDAR_SCANNER_INTERVAL  = "1m"
CONCOURSE_LIDAR_CHECKER_INTERVAL = "1m"
CONCOURSE_SECRET_CACHE_ENABLED  = true
CONCOURSE_SECRET_CACHE_DURATION  = "10m"
```

Is there any possibility to see the checks in the logs? Or configure them to be logged?
Thanks again!

**Reply by taylorsilva** [reply]
I think you can grep for `checker` in the logs and some stuff will show up. Not sure what the level of detail is but stuff definitely is logged. You could try changing the debug level to `debug` to ensure you're getting everything.

Can you give more details about the behaviour you saw? How long until polling stopped? How did you know polling was working and then stopped?

**Reply by ktreptow** [reply]
It seems to work now without changing anything, so the configuration above seems to be correct for my case. Thanks for your help!

On another note, how exactly does the `time` resource work? Let's say i have one that's configured like the following: 

```
- icon: clock-outline
  name: every-60s
  source:
    interval: 60s
  type: time
```

This Resource will update every 60 seconds but the checker, which is configured like the example above, will still only check every 5 minutes, correct?

**Reply by taylorsilva** [reply]
Yup you are correct! It will end up only getting checked every 5mins and that's when the time resource will see "has 60s passed since the previous version?"

**Reply by ktreptow** [reply]
Thanks a lot :) I Should still be able to configure `check_every: 1m` per certain resources, correct?

**Reply by taylorsilva** [reply]
Yes, if you want a certain resource checked more frequently then you can set `check_every` per resource to something else.

**Reply by ktreptow** [reply]
Somehow that does not seem to work as expected. The check ist performed between every 2,5 and 5 minutes bus is configure as every 1 minute. Is there anything to consider for this to work?

**Reply by taylorsilva** [reply]
Reading this again and I'm thinking I misunderstood what you meant by

> I Should still be able to configure check_every: 1m per certain resources, correct?

Also not sure what you meant by this sentence. Specifically not sure what you meany with the word "bus".

> The check ist performed between every 2,5 and 5 minutes bus is configure as every 1 minute

Resources are only checked as frequently as lidar is checking. if a resources `check_every` is less than lidar's checking interval, then the resource will NOT be checked more frequently. All checking is kicked-off by lidar. If lidar is not checking frequently enough for you then you need to reduce it's time. Overall I'd say it's better to keep lidar's check interval very short and selectively increase the `check_every` of resources. 

If you want, by default, resources to NOT be checked too frequently, then I would change `CONCOURSE_RESOURCE_CHECKING_INTERVAL` and `CONCOURSE_RESOURCE_TYPE_CHECKING_INTERVAL`. Leave the rest of the settings you mentioned at their default.

**Reply by ktreptow** [reply]
Oh, I meant to write "but" instead of "bus", sorry. 
 
Lidar is set to 1 minute. 
`CONCOURSE_RESOURCE_CHECKING_INTERVAL` and `CONCOURSE_RESOURCE_TYPE_CHECKING_INTERVAL` are both set to 5 minutes. 
My question was, if `check_every` should still work, when I set it to 1 minute? Because that isn't the case for me right now.

Sorry for the confusion and thanks very much for your help and patience!
